offchain/advance-runner/src/lib.rs

```diff
@@ -34,17 +34,25 @@ mod snapshot;
 pub async fn run(config: Config) -> Result<(), AdvanceRunnerError> {
     tracing::info!(?config, "starting advance runner");
 
-    let health_handle = http_health_check::start(config.health_check_config);
+    let health_handle = http_health_check::start(
+        config.health_check_config.runner_healthcheck_address,
+        config.health_check_config.runner_healthcheck_port,
+    );
     let advance_runner_handle =
         start_advance_runner(config.advance_runner_config);
 
-    tokio::select! {
-        ret = health_handle => {
-            ret.context(error::HealthCheckSnafu)
-        }
-        ret = advance_runner_handle => {
-            ret
+    if config.health_check_config.runner_enable_healthcheck == "true".to_owned()
+    {
+        tokio::select! {
+            ret = health_handle => {
+                ret.context(error::HealthCheckSnafu)
+            }
+            ret = advance_runner_handle => {
+                ret
+            }
```

Comment:
We will have problems with this `if` when we add the metrics_handle, but let me worry about this in my PR. Just letting you know this `if` is probably going to change.